src/clib/pio_file.c

```diff
@@ -200,8 +203,14 @@ int PIOc_openfile_retry(const int iosysid, int *ncidp, int *iotype,
 	if ((mpierr = MPI_Bcast(&file->mode, 1, MPI_INT, ios->ioroot, ios->union_comm)))
 	    return check_mpi(file, mpierr, __FILE__, __LINE__);
       
-	if ((mpierr = MPI_Bcast(&file->fh, 1, MPI_INT, ios->ioroot, ios->union_comm)))
+      tmp_fh = file->fh;
+	if ((mpierr = MPI_Bcast(&tmp_fh, 1, MPI_INT, ios->ioroot, ios->union_comm)))
```

Comment:
tmp_fh is used just below at line 212.  In pnetcdf or netcdf4p the filehandle is set on all iotasks and may differ from one task to another - this mechanism prevents it from being overwritten.